### ISSUE TITLE
[java] While and For statements followed by control statement

### ISSUE THREAD

**Issue by numeralnathan** [pmd/pmd#2261]
**Affects PMD Version:** 

6.20.0

**Rule:**

New rules: `WhileStatementFollowedByControlStatement` and `ForStatementFollowedByControlStatement`

**Description:**

It is my opinion that `while` and `for` should never be immediately followed by `break`, `continue`, `return` or `throw` control statements.  Instead, these control statements should be preceded by an `if`.

**Code Sample demonstrating the issue:**

```
   while (count > 5)
      break;                // This should be flagged as a mistake

   while (count > 5)
      continue;                // This should be flagged as a mistake

   while (count > 5)
      return;                // This should be flagged as a mistake

   while (count > 5)
      throw new IllegalStateException();                // This should be flagged as a mistake

   for (i = 0; i < 10; i++)
      break;                // This should be flagged as a mistake

   for (i = 0; i < 10; i++)
      continue;                // This should be flagged as a mistake

   for (i = 0; i < 10; i++)
      return;                // This should be flagged as a mistake

   for (i = 0; i < 10; i++)
      throw new IllegalStateException();                // This should be flagged as a mistake
```

**Comment by adangel** [comment]
I'd suggest to create a single rule for both loop statements. The rule can still provide properties, if one only wants to flag e.g. while-loops while allowing "breaks" for for-loops. IMHO the two rules are too similar to justify two completely separate rules. We did something like that with [ControlStatementBraces](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#controlstatementbraces).
